ThePhasesOfCraftship/2_best_practice_first/strategicDesignPart1/exercises/1_RefactoringTo4Tiers/begin/src/dto/AssignmentRequestDto.ts

```diff
@@ -0,0 +1,72 @@
+import { ErrorExceptionType } from "../constants"
+import { isUUID } from "../utils"
+
+export type AssignmentRequestDto = {
+ classId?: string
+ title?: string
+ id?: string  
+ studentId?: string
+ assignmentId?: string
+ grade?: 'A' | 'B' | 'C' | 'D'
+}
+
+export default function createAssingmentRequestDto(data: unknown) {
+  if(!data || typeof data !== 'object') {
+    throw new Error(ErrorExceptionType.ValidationError)
+  }
+  
+  const {classId , title, id, studentId, assignmentId, grade } = data as Partial<AssignmentRequestDto>
```

Comment:
Since all keys on the dto are optional already, using Partial is redundant.
```suggestion
  const {classId , title, id, studentId, assignmentId, grade } = data as AssignmentRequestDto
```

Reply:
Thank you @enekesabel for taking the time to review my code. I will try to level up my coding skills but also clarity levels to I don't make the mistakes I've done here in this PR

Reply:
You're welcome, I'm happy to help! :) None of these are major mistakes, I just like to be nitpicky :D